lib/aws/request.ex

```diff
@@ -42,22 +42,27 @@ defmodule AWS.Request do
     payload = encode!(client, metadata.protocol, input)
     headers = Signature.sign_v4(client, now(), "POST", url, headers, payload)
 
-    case AWS.Client.request(client, :post, url, payload, headers, options) do
-      {:ok, %{status_code: 200, body: body} = response} ->
-        body = if body != "", do: decode!(client, metadata.protocol, body)
-        {:ok, body, response}
+    telemetry_metadata = %{client: client, service: metadata, action: action, input: input}
 
-      {:ok, response} ->
-        {:error, {:unexpected_response, response}}
+    :telemetry.span([:aws, :request], telemetry_metadata, fn ->
+      case AWS.Client.request(client, :post, url, payload, headers, options) do
+        {:ok, %{status_code: 200, body: body} = response} ->
+          body = if body != "", do: decode!(client, metadata.protocol, body)
+          {{:ok, body, response}, %{}}
 
-      error = {:error, _reason} ->
-        error
-    end
+        {:ok, response} ->
+          {{:error, {:unexpected_response, response}}, %{}}
+
+        error = {:error, _reason} ->
+          {error, %{}}
+      end
+    end)
   end
 
   def request_rest(
         %Client{} = client,
         %ServiceMetadata{} = metadata,
+        action,
```

Comment:
This `action` does not exist for REST services (at least not available now). The closest thing to it sees to be conjunction of HTTP method and the path. WDYT of using that for the `action` field?

We could do something like:

```elixir
"#{http_method |> Atom.to_string() |> String.upcase()} #{path}"
```

Reply:
I've added those on new generated code.

The reason of why `action` is desirable, by [OpenTelemetry docs][1] (and some [reference implementation][2]) the spans should be named as `ServiceName.OperationName`.

Does that makes sense?

[1]: https://github.com/open-telemetry/opentelemetry-specification/blob/9216eb8ff1832ae790369e49ce326cb6bc3d1c80/specification/trace/semantic_conventions/instrumentation/aws-sdk.md#common-attributes
[2]: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/191a9a00cec2589fbf6fd5f68a586b54150cef04/instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/AwsSdkInstrumenterFactory.java#L46-L50